### PR TITLE
Document join-related optimizations

### PR DESCRIPTION
**Pull request by rschlussel-zz**
supercedes https://github.com/Teradata/presto/pull/424

Fixed up some of the English and added background and guidelines for manual join reordering

### CODE REVIEW

**Comment by rschlussel-zz** [comment]
FYI @pnowojski

**Review by kbajda — APPROVED** [review]

File: presto-docs/src/main/sphinx/optimizer/join-distribution-type.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
+********************************
+Partitioned and Replicated Joins
```
Comment (akshatnair) [inline presto-docs/src/main/sphinx/optimizer/join-distribution-type.rst]: this has to be Repartitioned

File: presto-docs/src/main/sphinx/optimizer/reorder-joins.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
+***************
+Join Reordering
+***************
+
+Background
+----------
+
+In a query with multiple joins, the order in which the joins are executed can have a significant impact on the query
```
Comment (akshatnair) [inline presto-docs/src/main/sphinx/optimizer/reorder-joins.rst]: It is applicable even for a single join scenario

**Review by akshatnair — COMMENTED** [review]
LGTM mod 2 small comments

**Comment by pnowojski** [comment]
lgtm

**Comment by rschlussel-zz** [comment]
merged

**Pull request closed without merging**